source/extensions/http/cache/file_system_http_cache/file_system_http_cache.cc

```diff
@@ -388,6 +404,108 @@ void FileSystemHttpCache::trackFileRemoved(uint64_t file_size) {
   stats_.size_bytes_.set(size_bytes_);
 }
 
+bool FileSystemHttpCache::needsEviction() const {
+  if (config().has_max_cache_size_bytes() &&
+      size_bytes_ > config().max_cache_size_bytes().value()) {
```

Comment:
it would seem easy to have two thresholds, and the one we check here should be higher.

Reply:
Four thresholds, because there's already two. I'm trying not to touch the API in this PR since then an API review is also required. More thresholds can be another change (again, I thought we liked small changes! More thresholds is in DESIGN.md as a goal now.)